api-v2/src/plan/plan.service.ts

```diff
@@ -0,0 +1,77 @@
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Student } from 'src/student/entities/student.entity';
+import { StudentService } from 'src/student/student.service';
+import { Repository } from 'typeorm';
+import { CreatePlanDto } from './dto/create-plan.dto';
+import { UpdatePlanDto } from './dto/update-plan.dto';
+import { Plan } from './entities/plan.entity';
+import { PlanNotFoundError } from './plan-not-found.error';
+
+@Injectable()
+export class PlanService {
+  constructor(
+    @InjectRepository(Plan)
+    private planRepository: Repository<Plan>,
+  ) {}
+
+  create(createPlanDto: CreatePlanDto, student: Student): Promise<Plan> {
+    const newPlan = this.planRepository.create({ ...createPlanDto, student });
+
+    try {
+      return this.planRepository.save(newPlan);
+    } catch (_error) {
+      throw new Error('Something went wrong when creating the new Plan');
+    }
+  }
```

Comment:
when would the save ever throw?

Reply:
save would throw if `newPlan` violates any table constraint(duplicate key, unique field, non nullable) etc.